metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/common/StellarCompiler.java

```diff
@@ -219,14 +220,23 @@ public Object apply(ExpressionState state) {
               //short circuit the if/then/else
               instanceDeque.pop();
               if((Boolean)curr.getValue()) {
-                //choose then
-                skipElse = true;
+                //choose then.  Need to make sure we're keeping track of nesting.
+                skipElseCount++;
               } else {
                 //choose else
+                // Need to count in case we see another if-else, to avoid breaking on wrong else.
+                int innerIfCount = 0;
                 while (it.hasNext()) {
                   Token<?> t = it.next();
+                  if (t.getUnderlyingType() == IfExpr.class) {
+                    innerIfCount++;
+                  }
```

Comment:
wouldn't if, else if be more clear here?

Reply:
I might have addressed this? If the current change isn't what you wanted, let me know.